### PR TITLE
Cert management

### PR DESCRIPTION
**Pull request by weinimo**
Seemed functioning before I did the rebase, but I couldn't fully test it yet.

### CODE REVIEW

File: controllers/amphoracontroller_controller.go [hunk 1]
```diff
@@ -401,11 +413,24 @@ func (r *OctaviaAmphoraControllerReconciler) generateServiceConfigMaps(
 	if err != nil {
 		return err
 	}
-	templateParameters["ServiceUser"] = instance.Spec.ServiceUser
+	caPassSecret, _, err := secret.GetSecret(
+		ctx, helper, instance.Spec.CAKeyPassphrasesSecret, instance.Namespace)
+	if err != nil {
+		return err
```
Comment (weinimo) [inline controllers/amphoracontroller_controller.go:419]: Not sure how to handle that case.

**Review by weinimo — COMMENTED** [review]
I will clean things up a squash the WIP commits later.

File: pkg/amphoracontrollers/volumes.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	configVolume = "amphora-certs"
+)
+
+var (
+	// Files get mounted as root:root, but process is running as octavia
+	configMode int32 = 0644
```
Comment (beagles) [inline pkg/amphoracontrollers/volumes.go:28]: I think this is okay for now but we should look at if its possible to mount as the octavia user.

**Review by beagles — APPROVED** [review]

File: pkg/amphoracontrollers/amphora_certs.go [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+	octaviav1 "github.com/openstack-k8s-operators/octavia-operator/api/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+var (
+	subjectDefault = pkix.Name{
+		Organization:  []string{"Dis"},
```
[the remark this answers is not in the document]
Comment (weinimo) [reply pkg/amphoracontrollers/amphora_certs.go:27]: It is what [TripleO used](https://opendev.org/openstack/tripleo-ansible/src/branch/master/tripleo_ansible/roles/octavia_overcloud_config/tasks/certs_gen.yml#L86). Not sure if it is needed for backwards compatibility, but I figured it is better to use what we know is working.

**Pull request merged**